src/KioskMode/README.md

```diff
@@ -0,0 +1,102 @@
+
+* A Test needs to be displayed in a restricted way for the use in an E-Exam.
+* The LTI-Interface needs the ability to display objects in a Kiosk-Mode when it
+  ILIAS is used as a Tool-Provider.
+* The Learning Sequence requires a kiosk mode to maintain control over the general
+  display and navigation.
+
+This library facilitates the implementation of objects that can be viewed in a kiosk
+mode and of features that display these objects in a kiosk mode. The provided means
+aim to make it possible for objects to implement a kiosk mode only once and then be
+used by all features that need to display objects in that kiosk mode. On the other
+hand, features that require viewing other objects in a kiosk mode should be able to
+use all objects that implement a kiosk mode.
+
+This library does neither implement the kiosk-mode for certain objects, nor implements
+a view on these objects. It rather provides interfaces that describe how consumers
+and providers of a kiosk mode should collaborate and functionality that helps to
+implement both sides of the collaboration.
+
+This library is complemented by the [Kiosk-Mode-Service](../../Services/KioskMode/README.md)
+that provides functionality to construct views for certain objects and concrete
+database functionality.
+
+## Architecture
+
+A **provider** of a kiosk mode is an ILIAS-object that may be viewed in in a kiosk
+mode. A **player** is any feature of ILIAS that wants to display ILIAS-objects in
+the reduced way defined by the kiosk mode.
+
+The **player** may request a **view** for a certain **provider** at the kiosk mode
+service. The **view** then is the exclusive way for the **player** and the
+**provider** to communicate.
+
+The **view** implements the following functionality to display and control the
+providing object via the kiosk mode:
+
+* It provides **controls** to the player that can be used to modify the state of
+  the view.
+* It provides a function that **updates** a **state** based on the use of **controls**
+  to track the condition of the view.
+* It provides a function to **render** the **state** to allow the player to display
+  the object in a kiosk mode.
+
+The **player** in turn uses that functions of the view to display and execute the
+kiosk mode of the object:
+
+* It asks the view for **controls** and displays them in an appropriate way.
+* It processes user input for the **controls** and asks the view to **update** the
+  **state**. It makes sure that the state is persisted appropriately.
+* It asks the view to **render** the **state** and displays the result in an
+  appropritate way and visual environment, most possibly in conjunction with the
+  controls.
+
+
+## Implementing a Provider
+
+Please use `ilKioskModeView` in `Services\KioskMode\classes\` as a base to implement
+this for ILIAS-objects.
+
+* `State` MUST only contain information about the view (not about LP, completion, ...)
```

Comment:
Two points:

- The state concept is used somehow inconsistent and hard to understand. Here it says clearly "not about LP" but the TOCBuilder is using constants like STATE_NOT_STARTED, STATE_IN_PROGRESS, STATE_COMPLETED, STATE_FAILED. Is this not LP vocabulary or is this workflow related.
- Are states attached to controls, subelements of controls or to views?
- Is this a big key value store without typisation inside and vulnerable to primitive obsession within its values? 
- Why does the Kiosk mode use this terms of other components (STATE_FAILED) and why is the kiok mode responsible to handle them at all? This is not needed in the LTI or SafeExam scenario. It may be a requirement of the learning sequences only and should be moved to the responsibility of this respective service then.